crypto/x509/by_dir.c

```diff
@@ -327,10 +327,10 @@ static int get_cert_by_subject(X509_LOOKUP *xl, X509_LOOKUP_TYPE type,
         /*
          * we have added it to the cache so now pull it out again
          */
-        CRYPTO_THREAD_write_lock(ctx->lock);
+        CRYPTO_THREAD_write_lock(xl->store_ctx->lock);
```

Comment:
@petrovr makes an excellent point, this should change to:

```
        X509_STORE_lock(xl->store_ctx);
```

Reply:
x509_lu.c has quite a few of these:

541: CRYPTO_THREAD_write_lock(ctx->ctx->lock); 
550: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
558: CRYPTO_THREAD_write_lock(ctx->ctx->lock); 
561: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
572: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
578: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
599: CRYPTO_THREAD_write_lock(ctx->ctx->lock); 
602: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
612: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
618: CRYPTO_THREAD_unlock(ctx->ctx->lock); 
693: CRYPTO_THREAD_write_lock(ctx->ctx->lock); 
720: CRYPTO_THREAD_unlock(ctx->ctx->lock);

Reply:
Yeah, ok.